src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java

```diff
@@ -387,4 +387,23 @@ public static TimeValue nodeTimeValue(Object node) {
             throw new ElasticsearchParseException(desc + " should be a hash but was of type: " + node.getClass());
         }
     }
+
+    /**
+     * Returns an array of string value from a node value.
+     *
+     * If the node represents an array the corresponding array of strings is returned.
+     * Otherwise the node is treated as a comma-separated string.
+     */
+    public static String[] nodeStringArrayValue(Object node) {
```

Comment:
I find the `node` wording confusing, it makes me think of elasticsearch nodes. Maybe `xContentNode`?